rootex/framework/systems/script_system.h

```diff
@@ -1,6 +1,7 @@
 #pragma once
 
 #include "system.h"
+#include "framework/entity_factory.h"
```

Comment:
This has no reason being here. Try to reduce as much includes as possible inside header files. This can be safely moved inside the .cpp file